logprep/runner.py

```diff
@@ -245,8 +260,13 @@ def stop(self):
             self._continue_iterating.value = False
 
     def _keep_iterating(self):
-        with self._continue_iterating.get_lock():
-            return self._continue_iterating.value
+        """generator function"""
+        while True:
+            with self._continue_iterating.get_lock():
+                iterate = self._continue_iterating.value
+                if not iterate:
+                    return
+                yield iterate
```

Comment:
with this change we could use a for loop instead of a while loop in the method `start()`. This leads to easier tests, because we have the control over how often the loop will happen by patching `runner._keep_iterating` with our own generator function

Reply:
and we get rid of the `StoppinIteration` error because the return in a generator function raises the `StopIteration` error. It is a more pythonic approach

Reply:
Ah okay, the for loop I have noticed but the advantage of the tests I didn't see yet. And also a neat behavior of a generator. Thanks for the explanation :+1: